allennlp/data/dataset.py

```diff
@@ -77,7 +77,7 @@ def get_padding_lengths(self) -> Dict[str, Dict[str, int]]:
 
     def as_arrays(self,
                   padding_lengths: Dict[str, Dict[str, int]] = None,
-                  verbose: bool = True) -> Dict[str, List[numpy.array]]:
+                  verbose: bool = True) -> Dict[str, Union[numpy.array, Dict[str, numpy.array]]]:
```

Comment:
It should, but it broke the type checking 😢 . I discussed this with @joelgrus  and we decided it wasn't worth the time figuring out why this was happening - it was something to do with using the `TypeVar` annotation from `typing`, because defining a new type just equal to this complex type didn't break mypy.